packages/@sanity/form-builder/src/inputs/common/fileTarget/fileTarget.tsx

```diff
@@ -122,7 +122,7 @@ export function fileTarget<ComponentProps>(Component: React.ComponentType<Compon
       (event: React.DragEvent) => {
         event.stopPropagation()
 
-        if (onFilesOver) {
+        if (onFilesOver && forwardedRef.current.isEqualNode(event.currentTarget)) {
```

Comment:
Could we have used `===` instead of `isEqualNode` here?

Reply:
Just went to double check documentation and `===` works just as well as ìsEqualNode` :) 
I'm going to do a couple of more tests and replace it if everything looks fine 👍